cmd/hook.go

```diff
@@ -49,13 +48,15 @@ func launchHook(hook string, timeout time.Duration, meta map[string]string) erro
 		return fmt.Errorf("start command: %w", err)
 	}
 
-	timer := time.AfterFunc(timeout, func() {
-		log.Println("hook timed out: killing command")
-		_ = cmd.Process.Kill()
-		_ = stdout.Close()
-	})
-
-	defer timer.Stop()
+	go func() {
+		select {
+		case <-ctxCmd.Done():
+			if ctxCmd.Err() != nil {
+				_ = cmd.Process.Kill()
+				_ = stdout.Close()
+			}
+		}
+	}()
```

Comment:
`gosimple` suggests something like

```suggestion
	go func() {
		<-ctxCmd.Done():
		if ctxCmd.Err() != nil {
			_ = cmd.Process.Kill()
			_ = stdout.Close()
		}
	}()
```